src/docs/components/ComponentBlock.jsx

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+class ComponentBlock extends Component {
+
+  constructor(props) {
+    super(props);
+    this.oldHTML = '';
+  }
+
+  componentDidMount() {
+    this.updateHTML();
+  }
+
+  componentDidUpdate() {
```

Comment:
We can go even step father in this case

Do it on componentWillReceiveProps, delete oldHTML or even state.oldHTML
compare props if there are equal (equal reference) if not check if equal inner html by: ReactDOMServer.renderToStaticMarkup method. 

In that case we could store this.oldHTML for caching only, for faster calculating.